packages/types/src/element/elementBox.ts

```diff
@@ -14,22 +14,6 @@
  * limitations under the License.
  */
 
-// TODO(#12126): Use improved Element type from shared package.
-export interface Element {
-  x: number;
-  y: number;
-  width: number;
-  height: number;
-  rotationAngle: number;
-  isBackground: boolean;
-  border: {
-    top: number;
-    right: number;
-    bottom: number;
-    left: number;
-  };
-}
-
 export interface ElementBox {
```

Comment:
Should this also be removed from here now?

Reply:
The file has been "moved" so it's no longer where it was.

Reply:
> To iterate on my previous comment, the main things standing out:
> 
> 1. Pattern types -> more suitable for `patterns` package
> 2. Specific element types (`ProductElement`, `GifElement`, etc.) -> more suitable for `element-library`
> 3. Resource types -> `media` package
> 
> There shouldn't be any circular dependencies that way.

I think it's a lot of work to do it that way, but doable if we're diligent.

Some of the utility functions inside the media package need access to element types (e.g. `generateVideoStrip(e:MediaElement, ...)`), but because the element library package would depend on the media package (the `VideoElement` needs to reference the `VideoResource`, etc), the media package cannot depend in reverse. So the media element would have to be defined in the media package.

That means that the abstract catch-all element type would exist in the types package, the abstract media element type would exist in the media package, and the concrete element types (some extending element, others extending media element) would exist in the element library package.

Sure, it would work, but it would also be a bit confusing in my opinion. Another option would be to move the `generateVideoStrip` function somewhere else, so the media element type can be defined in the element library package. But to where?

I also notice that the element library package depends on a ton of other packages, so none of those can ever depend on any specific element type, which would seem to hamper us a bit. E.g. the animation package, which have methods referring to media elements (e.g. in `mediaPositions.js`). Similarly inside the rich-text package, where `usePastedTextContent` needs to know about the text element interface.

I think the problem with this "types belong in their packages" approach is that we often combine both core functionality and broader utility functions inside a package - those broader utility functions often need types from other packages, but that creates a mess in dependencies.

While having all the types in a single package perhaps isn't best-practice, it would make things a lot simpler.

Would it make a difference if we made the more generic interfaces, of which actual elements would implement one or more? So we have `EditableElement`, `MediaElement`, `TrimmableElement`, `BackgroundableElement`, `ColorableElement`,  etc? Then functions through would accept one of those interfaces, and the actual elements could be passed because they implemented said interface?

Such a solution feels more "computer sciencey", but also more complex, as our actual elements are super concrete and such abstractions would be an almost useless exercise except to make the type system consistent.

WDYT?

Ping @miina and @merapi too.